File: 485-max-consecutive-ones/max-consecutive-ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int curr_sum = 0, max_sum = 0;

        for(auto num : nums){
            if(num == 1){
                curr_sum ++;
                max_sum = max(max_sum, curr_sum);
            }
            else{
                curr_sum = 0;
            }
        }
        return max_sum;
    }
};

File: 485-max-consecutive-ones/max-consecutive-ones.py
class Solution(object):
    def findMaxConsecutiveOnes(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        curr, max_sum = 0, 0

        for num in nums:
            if num == 1:
                curr += 1
                max_sum = max(curr, max_sum)
            else:
                curr = 0
        return max_sum
        